drivers/gpio/gpio_mchp_xec_v2.c

```diff
@@ -125,10 +135,12 @@ static int gpio_xec_configure(const struct device *dev,
 	 */
 	mask |= MCHP_GPIO_CTRL_PUD_MASK;
 
+	/* if both pulls are enabled the pin is in repeater mode */
 	if ((flags & GPIO_PULL_UP) != 0U) {
 		/* Enable the pull and select the pullup resistor. */
 		pcr1 |= MCHP_GPIO_CTRL_PUD_PU;
-	} else if ((flags & GPIO_PULL_DOWN) != 0U) {
+	}
+	if ((flags & GPIO_PULL_DOWN) != 0U) {
```

Comment:
this is changing GPIO driver behavior: also deserves a commit explaining why

Reply:
I am planning to focus only on pinctrl changes for this commit. So will remove changes not specific to pinctrl. 
Do we still need multiple commits, please let me know?

Reply:
Yes, you can take as a reference implementations already merged or in progress listed in https://github.com/zephyrproject-rtos/zephyr/issues/39740

Reply:
Thanks for the examples; will update this pull request with multiple commits for easy review.

Reply:
I have added the pinctrl changes as a series of 9 commits.